### PR TITLE
Create a config flow

### PR DESCRIPTION
**Pull request by mrueg**
Took a first stab and it's still work in progress.

* [x] How to set up the platform? Does async_create_platform exist?
* [x] Add missing stop name field 
* [x] Figure out how to query api and allow the user to select the right stop in a second step

### CODE REVIEW

**Comment by mrueg** [comment]
@vas3k this is almost ready to review, just need to fix the last linting issue. Feel free to take a look already.

**Comment by mrueg** [comment]
@vas3k this is ready for review

File: custom_components/berlin_transport/sensor.py [hunk 1]
```diff
@@ -38,42 +43,53 @@
...
             {
-                vol.Required(CONF_DEPARTURES_NAME): str,
-                vol.Required(CONF_DEPARTURES_STOP_ID): int,
-                vol.Optional(CONF_DEPARTURES_DIRECTION): int,
-                vol.Optional(CONF_DEPARTURES_DURATION): int,
-                vol.Optional(CONF_DEPARTURES_WALKING_TIME, default=1): int,
-                vol.Optional(CONF_SHOW_API_LINE_COLORS, default=False): bool,
+                vol.Required(CONF_DEPARTURES_NAME): cv.string,
+                vol.Required(CONF_DEPARTURES_STOP_ID): cv.positive_int,
+                vol.Optional(CONF_DEPARTURES_DIRECTION): cv.string,
```
Comment (mrueg) [inline custom_components/berlin_transport/sensor.py:61]: It seems like the Direction is a string and not an int, so I fixed that here https://v6.vbb.transport.rest/api.html#get-stopsiddepartures

File: custom_components/berlin_transport/config_flow.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    }
+)
+
+NAME_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_SEARCH): cv.string,
+    }
+)
+
+# IPv6 is broken, see: https://github.com/public-transport/transport.rest/issues/20
```
Comment (vas3k) [inline custom_components/berlin_transport/config_flow.py:53]: That's funny 🤣  I think I had these issues without realizing that

**Review by vas3k — APPROVED** [review]
That's really cool. I don't have anything to add on the code side, but really want to test how the whole flow will work in a real life.

And thanks for taking the time to do it. If this all really works, I'll send you a beer 🍺

File: custom_components/berlin_transport/config_flow.py (hunk 2 above)
Comment (mrueg) [reply custom_components/berlin_transport/config_flow.py:53]: Took me a bit to figure out why the request took so long in home-assistant while it worked quickly when using the browser. Then I checked with curl and saw it timing out on ipv6 :D

**Pull request merged**